ports/stm32/dma.c

```diff
@@ -448,10 +448,25 @@ void dma_init(DMA_HandleTypeDef *dma, const dma_descr_t *dma_descr, void *data){
         } else {
             // only necessary initialization
             dma->State = HAL_DMA_STATE_READY;
+
 #if defined(MCU_SERIES_F4) || defined(MCU_SERIES_F7)
             // calculate DMA base address and bitshift to be used in IRQ handler
             extern uint32_t DMA_CalcBaseAndBitshift(DMA_HandleTypeDef *hdma);
             DMA_CalcBaseAndBitshift(dma);
+
+#elif defined(MCU_SERIES_L4)
+            // copied from stm32l4xx_hal_dma.c:dma_init()
```

Comment:
Unfortunately copying code like this leads to licensing/copyright issues.... it's a small bit of code, can you rewrite it in your own way?  It would also be good to add a comment here about why this re-init of the fields needs to be done.